### ISSUE TITLE
UK-layout \ key is not recognised (# recognised as \)

### ISSUE THREAD

**Issue by dsalt** [00-Evan/shattered-pixel-dungeon#1873]
Trying to bind `\` to a function results in `New binding: None`. On confirming, any existing binding in that slot is _removed_.

The keymap (for testing purposes) is “English (UK, extended, Windows)” (with useful-for-me options for switching layout and Compose), which `setxkbmap -print` describes as:
```
xkb_keymap {
	xkb_keycodes  { include "evdev+aliases(qwerty)"	};
	xkb_types     { include "complete"	};
	xkb_compat    { include "complete"	};
	xkb_symbols   { include "pc+gb(extd)+gb(extd):2+inet(evdev)+group(shifts_toggle)+level3(ralt_switch)+compose(lwin-altgr)"	};
	xkb_geometry  { include "pc(pc105)"	};
};
```

(Looks like this is a library issue which needs to be forwarded.)

**Comment by 00-Evan** [comment]
Unfortunately as you mention this is likely an issue with keyboard handling on Shattered's game library, and not an issue with Shattered itself, consider reporting this to libGDX if you want. From a quick test on my end entering the '\' key does correctly show that as being bound, So it's possible that the game is receiving some other key code that it doesn't support, and so is showing 'none'.

Also, I'm afraid I'm not really able to test, debug, and ensure functionality for all the various different flavors and configurations of Linux.

I'm happy to keep discussing this but for now I'm closing this issue as it seems unlikely there is anything I can do directly to test/remedy it.

**Comment by dsalt** [comment]
Which key do you mean by “the '' key”? If you mean `\` (usually between Backspace and Return), that's `#` here (and it's between `'` and Return) – `\` is between Left Shift and `Z`.

Anyway, https://github.com/libgdx/libgdx/issues/7339 is about this. I see two pull requests, either of which should fix it; looks like we wait for now.

**Comment by 00-Evan** [comment]
The specific key isn't really what I meant, moreso that the game is accepting the '\\' keycode just fine, so something is causing either the OS to not send that, or the game library to not properly interpret the keycode as '\\', hopefully that can be amended in a future libGDX version.

**Comment by dsalt** [comment]
Well. It **is** about these keys, specifically.

`\` (US layouts) is the same key as `#` (UK layouts): same key code, different mapping. This key is recognised by libGDX as `\`.

`\` (UK layout) is normally *not present* on a US-layout keyboard. This key is not currently handled by libGDX.

I've been able to build a minimally-patched libGDX 1.12.1 (https://github.com/libgdx/libgdx/pull/7456 modified to add names for `WORLD_1` and `WORLD_2`) and relinked SPD against it (messing with gradle's cache…). With that in place, `\` (UK layout) is recognised and is bindable.

Hopefully, 1.12.2 will have this fixed.
